src/vehicle.cpp

```diff
@@ -6845,11 +7056,88 @@ bounding_box vehicle::get_bounding_box()
     return b;
 }
 
+bool vehicle::fake_part( int part_num ) const
+{
+    for( const auto &fm : fake_mounts ) {
+        if( fm.second.fake_part_index == part_num ) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool vehicle::valid_part( int part_num, bool include_fake ) const
+{
+    return part_num >= 0 && part_num < ( include_fake ?  num_parts_incl_fake() :  num_parts() );
+}
+
+bool vehicle::has_any_parts() const
+{
+    return static_cast<int>( parts.size() ) > 0;
+}
+
+int vehicle::num_parts() const
+{
+    return static_cast<int>( parts.size() ) - fake_part_count;
+}
+
+int vehicle::num_parts_incl_fake() const
+{
+    return static_cast<int>( parts.size() );
+}
+const vehicle_part &vehicle::cpart( int part_num ) const
+{
+    return const_cast<vehicle_part &>( parts[part_num] );
```

Comment:
```suggestion
    return parts[part_num];
```

Reply:
I honestly appreciate the effort. I had forgotten this PR was still open. Once I get some more free time I will have a look at your suggestions and create a PR with only the fake parts changes.